src/extras/module-types.js

```diff
@@ -6,90 +6,90 @@
-        });
-    }
-    return instantiate.apply(this, arguments);
-
-    function getSourceRes () {
-      return fetch(url).then(function (res) {
+    if (this.shouldFetch(url)) {
+      return this.fetch(url)
+      .then(function (res) {
         if (!res.ok)
-          loadError(res.status + ' ' + res.statusText);
-        return res;
+          throw Error(res.status + ' ' + res.statusText + ', loading ' + url + (parent ? ' from ' + parent : ''));
+        const contentType = res.headers.get('content-type');
+        if (contentType.match(/^(text|application)\/(x-)?javascript(;|$)/)) {
+          return res.text().then(function (source) {
+            (0, eval)(source);
+            return loader.getRegister();
+          });
+        }
+        else if (contentType.match(/^application\/json(;|$)/)) {
+          return res.text().then(function (source) {
+            return [[], function (_export) {
+              return {
+                execute: function () {
+                  _export('default', JSON.parse(source));
+                }
+              };
+            }];
+          });
+        }
+        else if (contentType.match(/^text\/css(;|$)/)) {
+          return res.text().then(function (source) {
+            return [[], function (_export) {
+              return {
+                execute: function () {
+                  // Relies on a Constructable Stylesheet polyfill
+                  const stylesheet = new CSSStyleSheet();
+                  stylesheet.replaceSync(source);
+                  _export('default', stylesheet);
+                }
+              };
+            }];
+          }); 
+        }
+        else if (contentType.match(/^application\/wasm(;|$)/)) {
+          return (WebAssembly.compileStreaming ? WebAssembly.compileStreaming(res) : res.arrayBuffer().then(WebAssembly.compile))
+          .then(function (module) {
+            const deps = [];
+            const setters = [];
+            const importObj = {};
+        
+            // we can only set imports if supported (eg early Safari doesnt support)
+            if (WebAssembly.Module.imports)
+              WebAssembly.Module.imports(module).forEach(function (impt) {
+                const key = impt.module;
+                if (deps.indexOf(key) === -1) {
+                  deps.push(key);
+                  setters.push(function (m) {
+                    importObj[key] = m;
+                  });
+                }
+              });
+        
+            return [deps, function (_export) {
+              return {
+                setters: setters,
+                execute: function () {
+                  return WebAssembly.instantiate(module, importObj)
+                  .then(function (instance) {
+                    _export(instance.exports);
+                  });
+                }
+              };
+            }];
+          });
+        }
+        else {
+          throw new Error('Unknown module type "' + contentType + '"');
```

Comment:
With strict MIME checking, this now catches all unknown MIMEs, including the HTML error.